hack/hook-me.sh

```diff
@@ -0,0 +1,260 @@
+    imagePullPolicy: IfNotPresent
+    name: debug
+    resources:
+      limits:
+        cpu: 50m
+        memory: 128Mi
+      requests:
+        cpu: 20m
+        memory: 64Mi
+  volumes:
+  - name: inlets-tls
+    secret:
+      secretName: gardener-extension-webhook-cert
+  dnsPolicy: ClusterFirst
+  enableServiceLinks: true
+  restartPolicy: Always
+
+EOF
+}
+
+waitUntilPodIsReady(){
+    namespace=${1:-}
+    [[ -z $namespace ]] && echo "Please specify extension namespace!" && exit 1
+
+    until test "$(kubectl -n $namespace get pods inlets-server --no-headers | awk '{print $2}')" = "3/3"
+    do
+      sleep 2s
+    done
+}
+
+cleanUP() {
+   namespace=${1:-}
+   [[ -z $namespace ]] && echo "Please specify the extension namespace!" && exit 1
+
+   kubectl -n $namespace wait --for=delete svc/inlets-lb --timeout=60s
+
+   kill -9 $(pgrep inlets)
+   exit 0
+}
+
+usage(){
+  echo "===================================PRE-REQs========================================="
+  echo "\`host\` commands for DNS"
+  echo "\`inlets\` command. For mac, simply use \`brew install inlets\`, for linux \`curl -sLS https://get.inlets.dev | sudo sh\`"
+  echo "===================================================================================="
+
+  echo ""
+
+  echo "========================================================USAGE======================================================================"
+  echo "> ./hack/hook-me.sh <extension namespace e.g. extension-provider-aws-fpr6w> <provider e.g., aws>  <webhookserver port e.g., 8443>"
+  echo "> \`make EXTENSION_NAMESPACE=<extension namespace e.g. extension-provider-aws-fpr6w> start-provider-<provider-name e.g.,aws>-local\`"
+  echo "=================================================================================================================================="
+
+  echo ""
+
+  echo "===================================CLEAN UP COMMANDS========================================="
+  echo "> kubectl -n garden scale deploy gardener-resource-manager --replicas=1"
+  echo "> kubectl -n \$namespace scale deploy gardener-extension-provider-\$providerName --replicas=1"
+  echo "> kubectl -n \$namespace wait --for=delete svc/inlets-lb --timeout=60s"
+  echo "============================================================================================="
+
+  exit 0
+}
+if [[ "${BASH_SOURCE[0]}" = "$0" ]]; then
+
+  if [ "$1" == "-h" ] ; then
+        usage
+  fi
+
+  namespace=${1:-}
+  [[ -z $namespace ]] && echo "Please specify the extension namespace!" && exit 1
+
+  providerName=${2:-}
+  [[ -z $providerName ]] && echo "Please specify the provider name (aws,gcp,azure,..etc.)!" && exit 1
+
+  webhookServerPort=${3:-}
+  [[ -z $webhookServerPort ]] && echo "Please specify webhook server port" && exit 1
+
+
+  trap 'cleanUP $namespace' SIGINT SIGTERM
+
+  echo "[STEP 0] Ignore Updates for the webhook service and the deployment"
+  kubectl -n garden scale deploy gardener-resource-manager --replicas=0
```

Comment:
AFAIK, the `gardener-resource-manager` is scaled up again after some time by the `seed` control loop and reverts the changes you made to the seed's resources. Doesn't that break the `inlets` setup?

Furthermore, this `gardener-resource-manager` instance is used to deploy control plane components for shoot cluster. For example, the `extension-shoot-cert-service` makes the `gardener-resource-manager` take care about the following resources: https://github.com/gardener/gardener-extensions/tree/master/controllers/extension-shoot-cert-service/charts/internal/shoot-cert-management-seed/templates

This also breaks the end-to-end test scenario, doesn't it?

Reply:
Depends on how you use it, my use-case so far is to reconcile and introduce changes and let the new (local extension) handle the reconciliation and see it reconcile successfully, of course there are other use-cases as you mention where you want to start reconciling a shoot from scratch using the local-extension. In that case, we should not switch-off the resource manager. 

Another solution is to modify the charts to make use of the managed resource annotation. Let's discuss what would most make sense here.

Reply:
> Another solution is to modify the charts to make use of the managed resource ignore annotation. Let's discuss what would most make sense here.

As discussed, this may be the only viable solution when end-to-end testing is desired. It'd be nice to see a documentation explaining those prerequisites because it's not trivial to understand the rationale behind it.

Reply:
Can't we simply expose a new Helm chart value that, if set to `true`, adds the ignore annotation to the service behind the webhook pod? Then, if you want to run the webhook locally, you just need to start this script and change the mentioned flag in the respective `ControllerRegistration`. When you are done testing/running the webhook locally then you can easily go back to the standard deployment by removing the mentioned flag from the `ControllerRegistration` again.
No need to scale down the GRM.